### ISSUE TITLE
Preset icons are sometimes stuck loading

### ISSUE THREAD

**Issue by EvanHahn** [digidem/comapeo-mobile#611]
Sept 19 Update : Xime : Since the CoMapeo1.0.0-RC.3 build, it has been happening more consistently. See videos here : https://www.notion.so/digidem/Inconsistent-icons-share-behavior-1041b08162d580ae9714e9b406dfe905

Hopefully fixing it will also fix : https://www.notion.so/digidem/Share-difficulty-1041b08162d5801a9cc9eb9d2d396a82
***

**Describe the bug**

Preset icons don't seem to be appearing in multiple places. They are stuck in the loading state.

**Screenshots**

| ![Screenshot 1](https://github.com/user-attachments/assets/e84fbdb5-1b3e-4bcb-9c62-4a54e541932a) | ![Screenshot 2](https://github.com/user-attachments/assets/d2688207-bab3-410b-9ed9-160a6272cf6d) |
|---|---|

**To Reproduce**

I don't know how to reproduce this, but it seems to happen when joining a project for the first time.

It seems to go away when resetting the app.

**Additional context**

@achou11 saw this on one of his devices, @rudokemper on another, and @EvanHahn on a third.

I assume the bug comes from [this code](https://github.com/digidem/comapeo-mobile/blob/0da8ea773ad0d7d399abfe11b7c3eaf12cf2926f/src/frontend/hooks/server/presets.ts#L39-L60), but I'm not sure.

**Comment by jencastrodoesstuff** [comment]
I've added a few other videos to the drive, linked in the notion page in case those are helpful. (in those videos different alt icons  sometimes appeared including the black location drop, and the red warning triangle.)

**Comment by achou11** [comment]
One of the primary issues is that opening the file picker causes the app to pause and resume, which causes the HTTP server that serves the assets like icons to restart. Relevant code here:

https://github.com/digidem/comapeo-mobile/blob/9b121f0a64f00fb7f3c2339d66bcf91fb9e3dd3c/src/backend/src/app.js#L115-L124

To test the timing, I updated the resume as following:

```js
console.time('resume')
fastifyController.start().then(() => {
  console.timeEnd('resume')
})
```

when going through the config import flow on a relatively lower end device (old-ish Pixel 2) and importing the wampis example, it took **almost 1 minute** for the server to start up again, which is very slow. During that time, if you try to go to any screen that load an HTTP-served assets (such as the category icons), we're attempting to get the server url that points to the asset but it take too long before we decide to quit, which falls back to a generic place icon. if you navigate in and out in the right ways (in dev terms: basically making sure the relevant component unmounts), then you'll most likely eventually get the url that points to the image, which can then eventually load the image as expected.

Interestingly, going through the same import flow but for the **default config** takes much less time for the server to resume (a few milliseconds). Given this, my guess is that the server start time is dependent/blocked by either:

  - the config import itself
  - the indexing that happens after importing a config

---

EDIT 1: Upon more testing, I think it's only the **first** import of a substantial config that causes a long startup time. Some notes about the following sequence:

1. Import notable config = takes long time (~1 minute)
2. Import default config = takes little time (few milliseconds)
3. Import notable config from (1) = takes little time (few milliseconds)

Suggests that importing new entries that don't already exist in the database (i.e. after step 1) might cause something to take a long time, which subsequently affects something that's related to the server startup time.

---

EDIT 2: I don't what i'm talking about 😂 . with some help from Gregor, we noticed that **closing** the server can take a long time (around 40 seconds) because of the built-in behavior between http clients and the node http server. next thing to try is to force close the node server connections and see if that causes the keepalive to not happen.

---
 
As an aside, not sure using a stable port number (which is what was recommended as a potential remedy) will fix this specific issue. It could probably help for a separate issue related to avoiding stale urls, which reduces some complexity in frontend code.
